diff --git a/src/Plugin/Block/LBKBaseComponent.php b/src/Plugin/Block/LBKBaseComponent.php
index 3764d60..7f13d3b 100644
--- a/src/Plugin/Block/LBKBaseComponent.php
+++ b/src/Plugin/Block/LBKBaseComponent.php
@@ -83,6 +83,7 @@ abstract class LBKBaseComponent extends BlockBase
     $this->setConfiguration($configuration);
     $this->currentRouteMatch = $currentRouteMatch;
     $this->entityTypeBundleInfo = $entityTypeBundleInfo;
+    $this->baseId = "lbk_base";
   }
 
   /**
@@ -439,4 +440,11 @@ abstract class LBKBaseComponent extends BlockBase
     return $machineName;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinition($derivative_id) {
+    // Since this block plugin has no derivatives, we simply return null.
+    return null;
+  }
 }
diff --git a/src/Plugin/Block/LBKBookNavigation/LBKBookNavigation.php b/src/Plugin/Block/LBKBookNavigation/LBKBookNavigation.php
index 31a035c..e6063e0 100755
--- a/src/Plugin/Block/LBKBookNavigation/LBKBookNavigation.php
+++ b/src/Plugin/Block/LBKBookNavigation/LBKBookNavigation.php
@@ -298,32 +298,4 @@ class LBKBookNavigation extends LBKBaseComponent implements ContainerFactoryPlug
     }
     return $bookDescription;
   }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginDefintion() {
-    return $this->pluginDefinition;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginId() {
-    return $this->pluginId;
-    }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getBaseId() {
-    return $this->baseId;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getDerivativeId() {
-    return $this->derivativeId;
-  }
 }
diff --git a/src/Plugin/Block/LBKIconText/LBKIconText.php b/src/Plugin/Block/LBKIconText/LBKIconText.php
index b1e3acf..de0a9b1 100755
--- a/src/Plugin/Block/LBKIconText/LBKIconText.php
+++ b/src/Plugin/Block/LBKIconText/LBKIconText.php
@@ -249,32 +249,4 @@ class LBKIconText extends LBKBaseComponent implements ContainerFactoryPluginInte
 
     return $build;
   }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginDefintion() {
-    return $this->pluginDefinition;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginId() {
-    return $this->pluginId;
-    }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getBaseId() {
-    return $this->baseId;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getDerivativeId() {
-    return $this->derivativeId;
-  }
 }
diff --git a/src/Plugin/Block/LBKImage/LBKImage.php b/src/Plugin/Block/LBKImage/LBKImage.php
index 41cf0e8..e959508 100755
--- a/src/Plugin/Block/LBKImage/LBKImage.php
+++ b/src/Plugin/Block/LBKImage/LBKImage.php
@@ -258,32 +258,4 @@ class LBKImage extends LBKBaseComponent implements ContainerFactoryPluginInterfa
 
     return $build;
   }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginDefintion() {
-    return $this->pluginDefinition;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginId() {
-    return $this->pluginId;
-    }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getBaseId() {
-    return $this->baseId;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getDerivativeId() {
-    return $this->derivativeId;
-  }
 }
diff --git a/src/Plugin/Block/LBKRender/LBKRender.php b/src/Plugin/Block/LBKRender/LBKRender.php
index 15f4e98..61f1c69 100644
--- a/src/Plugin/Block/LBKRender/LBKRender.php
+++ b/src/Plugin/Block/LBKRender/LBKRender.php
@@ -328,32 +328,4 @@ class LBKRender extends LBKBaseComponent implements ContainerFactoryPluginInterf
 
     return $viewModes;
   }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginDefintion() {
-    return $this->pluginDefinition;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginId() {
-    return $this->pluginId;
-    }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getBaseId() {
-    return $this->baseId;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getDerivativeId() {
-    return $this->derivativeId;
-  }
 }
diff --git a/src/Plugin/Block/LBKRichText/LBKRichText.php b/src/Plugin/Block/LBKRichText/LBKRichText.php
index 2bf631a..a0eeeee 100755
--- a/src/Plugin/Block/LBKRichText/LBKRichText.php
+++ b/src/Plugin/Block/LBKRichText/LBKRichText.php
@@ -128,31 +128,4 @@ class LBKRichText extends LBKBaseComponent implements ContainerFactoryPluginInte
 
     return $build;
   }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginDefintion() {
-    return $this->pluginDefinition;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginId() {
-    return $this->pluginId;
-    }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getBaseId() {
-    return $this->baseId;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getDerivativeId() {
-    return $this->derivativeId;
-  }}
+}
diff --git a/src/Plugin/Block/LBKTab/LBKTab.php b/src/Plugin/Block/LBKTab/LBKTab.php
index 958404e..b63a41f 100755
--- a/src/Plugin/Block/LBKTab/LBKTab.php
+++ b/src/Plugin/Block/LBKTab/LBKTab.php
@@ -742,32 +742,4 @@ class LBKTab extends LBKBaseComponent implements ContainerFactoryPluginInterface
 
     return $allFields;
   }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginDefintion() {
-    return $this->pluginDefinition;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginId() {
-    return $this->pluginId;
-    }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getBaseId() {
-    return $this->baseId;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getDerivativeId() {
-    return $this->derivativeId;
-  }
 }
diff --git a/src/Plugin/Block/LBKVideo/LBKVideo.php b/src/Plugin/Block/LBKVideo/LBKVideo.php
index 0e66b94..a00f3e5 100644
--- a/src/Plugin/Block/LBKVideo/LBKVideo.php
+++ b/src/Plugin/Block/LBKVideo/LBKVideo.php
@@ -679,32 +679,4 @@ class LBKVideo extends LBKBaseComponent implements ContainerFactoryPluginInterfa
 
     return $machineName;
   }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginDefintion() {
-    return $this->pluginDefinition;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getPluginId() {
-    return $this->pluginId;
-    }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getBaseId() {
-    return $this->baseId;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getDerivativeId() {
-    return $this->derivativeId;
-  }
 }
