--- a/core/modules/views/src/Plugin/views/query/QueryPluginBase.php
+++ b/core/modules/views/src/Plugin/views/query/QueryPluginBase.php
@@ -149,7 +149,12 @@
    * Set an OFFSET on the query, specifying a number of results to skip.
    */
   public function setOffset($offset) {
-    $this->offset = $offset;
+    if (is_null($offset)) {
+      $this->offset = 0;
+    }
+    else {
+      $this->offset = $offset;
+    }
   }

   /**
